packages/aws-cdk/test/diff.test.ts

```diff
@@ -9,126 +9,254 @@ import { instanceMockFrom, MockCloudExecutable } from './util';
+          AdditionChild: {
+            Type: 'AWS::CloudFormation::Stack',
+            Resources: {
+              SomeResource: {
+                Type: 'AWS::Something',
+                Properties: {
+                  Prop: 'added-value',
+                },
+              },
+            },
+          },
+          DeletionChild: {
+            Type: 'AWS::CloudFormation::Stack',
+            Resources: {
+              SomeResource: {
+                Type: 'AWS::Something',
+              },
+            },
+          },
+          ChangedChild: {
+            Type: 'AWS::CloudFormation::Stack',
+            Resources: {
+              SomeResource: {
+                Type: 'AWS::Something',
+                Properties: {
+                  Prop: 'new-value',
+                },
+              },
+            },
+          },
+        };
+        return Promise.resolve({
+          Resources: {
+            AdditionChild: {
+              Type: 'AWS::CloudFormation::Stack',
+              Resources: {
+                SomeResource: {
+                  Type: 'AWS::Something',
+                },
+              },
+            },
+            DeletionChild: {
+              Type: 'AWS::CloudFormation::Stack',
+              Resources: {
+                SomeResource: {
+                  Type: 'AWS::Something',
+                  Properties: {
+                    Prop: 'value-to-be-removed',
+                  },
+                },
+              },
+            },
+            ChangedChild: {
+              Type: 'AWS::CloudFormation::Stack',
+              Resources: {
+                SomeResource: {
+                  Type: 'AWS::Something',
+                  Properties: {
+                    Prop: 'old-value',
+                  },
+                },
+              },
+            },
+          },
+        });
+      }
+      return Promise.resolve({});
+    });
   });
 
-  // THEN
-  expect(exitCode).toBe(1);
-});
+  test('diff can diff nested stacks', async () => {
+    // GIVEN
+    const buffer = new StringWritable();
+
+    // WHEN
+    const exitCode = await toolkit.diff({
+      stackNames: ['Parent'],
+      stream: buffer,
+    });
 
-test('throws an error during diffs on stack with error metadata', async () => {
-  const buffer = new StringWritable();
+    // THEN
+    const plainTextOutput = buffer.data.replace(/\x1B\[[0-?]*[ -/]*[@-~]/g, '');
+    expect(plainTextOutput.trim()).toEqual(`Stack Parent
+Resources
+[~] AWS::CloudFormation::Stack AdditionChild 
+ └─ [~] Resources
+     └─ [~] .SomeResource:
+         └─ [+] Added: .Properties
+[~] AWS::CloudFormation::Stack DeletionChild 
+ └─ [~] Resources
+     └─ [~] .SomeResource:
+         └─ [-] Removed: .Properties
+[~] AWS::CloudFormation::Stack ChangedChild 
+ └─ [~] Resources
+     └─ [~] .SomeResource:
+         └─ [~] .Properties:
+             └─ [~] .Prop:
+                 ├─ [-] old-value
+                 └─ [+] new-value`);
```

Comment:
I could see adding a doubly-nested Stack to this test. But that's optional, that is a nice test as-is.

Reply:
:+1:, but I think we should add the doubly-nested stack to this test in the follow-up PR that improves the formatting of the diff. (we'll have to change this test anyway at that point, so we should save some effort now).